src/vsr/grid.zig

```diff
@@ -611,29 +652,14 @@ pub fn GridType(comptime Storage: type) type {
             );
 
             if (grid.canceling) |_| {
+                assert(grid.write_queue.empty());
+
                 grid.write_iops.release(iop);
                 grid.cancel_join_callback();
                 return;
             }
 
-            if (completed_write.repair) {
-                // We wait until the write completes to resolve the repair queue, to prevent
-                // these writes from ever overlapping with compaction or checkpoints.
-                const header = schema.header_from_block(cache_block.*);
-
-                var read_ = grid.read_global_queue.peek();
-                while (read_) |read| : (read_ = read.next) {
-                    if (read.checksum == header.checksum and
-                        read.address == completed_write.address)
-                    {
-                        grid.read_global_queue.remove(read);
-                        grid.read_block_resolve(read, .{ .valid = cache_block.* });
-                        break;
-                    }
-                } else {
-                    unreachable;
-                }
-            }
+            if (completed_write.repair) grid.repair_queue.repair_complete(cache_block.*);
```

Comment:
Maybe move this to after 

```zig
completed_write.callback(completed_write);
```

?

`repair_complete` internally also invokes user-supplied callback, and I think we want to sequence that after all other effects of a write are carried out. 

I don't understand specifics here yet, but it also seems smelly that we know essentially attach two different callbacks to the write...